camunda-utils/Camunda/Camunda.py

```diff
@@ -1,37 +1,144 @@
     def set_output_variable(self, name, value):
-        """
-        Stores the value in-memory with the name as key.
+        """Sets the output variable of the Task in the calling process.
+
+        :param name: The name of the output variable.
+        :param value: The value of the output variable.
+
+        Examples:
+
+        .. code-block:: robotframework
+
+            # Set the output variable 'result' to the value 'Hello, World!'
+            Set Output Variable    result    Hello, World!
+
+            # Set output variable 'result' to the value of the variable ${output}
+            Set Output Variable    result    ${output}
         """
         self.outputs[name] = value
 
-    @keyword(name="Set Output File")
-    def set_output_file(self, name, path):
+    @keyword
+    def upload_documents(self, glob: str, variableName: Optional[str] = None):
+        """Upload a document to Camunda. Optionaly store the file descriptor in a process variable.
+
+        :param glob: A glob pattern with files to upload.
+        :param variableName: The name of the variable to store the file descriptor in.
+        :return: A file descriptor or a list of file descriptors of the uploaded documents.
+
+        Examples:
+
+        .. code-block:: robotframework
+
+            # Upload a single file
+            Upload Document    path/to/file.txt
+            Set Output Variable    fileDescriptor    ${fileDescriptor}
+
+            # Directly store the file descriptor in a variable
+            Upload Document    path/to/file.txt
+
+            # Upload all files in a directory
+            Upload Document    path/to/directory/*   variableName="invoices"
+
+            # Upload all .pdf files in the workspace
+            Set Output Variable    fileDescriptor    variableName="invoices"
         """
-        Parses the file at `path` using `create_file_object` and stores the return value in-memory with the name as a key.
+        url = f"{self.base_url}/file/store/{self.workspace_id}"
+        headers = {"Content-Type": "application/json"}
+
+        data = {"files": glob}
+
+        response = requests.post(url, headers=headers, data=json.dumps(data))
+
+        if response.status_code != 200:
+            response.raise_for_status()
+
+        fileDescriptors = list(response.json().values())
+
+        # If we only have 1 file, return the file descriptor as a string
+        if len(fileDescriptors) == 1:
+            fileDescriptors = fileDescriptors[0]
+
+        if variableName:
+            self.outputs[variableName] = fileDescriptors
+
+        return fileDescriptors
+
+    @keyword
+    def download_documents(self, fileDescriptor, path: Optional[str] = "") -> List[str]:
+        """Retrieve one or multiple documents from the backend.
+
+        :param fileDescriptor: The file descriptor of the document to retrieve.
+        :param path: The path where the document should be saved to. Defaults to root directory.
```

Comment:
I'd avoid the term **root directory** as that might be interpreted as the actual root directory (eg `/` or `C:\`). Maybe **root of the job workspace** or just **workspace directory**?

Reply:
Let's go with workspace directory